### PR TITLE
pep8 style fixes, no functional changes

### PR DESCRIPTION
**Pull request by msink**

### CODE REVIEW

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -1275,8 +1275,8 @@ def generate_dynamic_link_rules(self, outfile):
                 else:
                     command_template = ' command = %s %s $ARGS  %s $in $LINK_ARGS $aliasing\n'
                 command = command_template % \
-                (' '.join(compiler.get_linker_exelist()),\
-                 ' '.join(cross_args),\
+                (' '.join(compiler.get_linker_exelist()),
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[flake8]__
* [E122] continuation line missing indentation or outdented

File: mesonbuild/interpreterbase.py [hunk 2]
```diff
@@ -403,19 +403,19 @@ def bool_method_call(self, obj, method_name, args):
         (posargs, _) = self.reduce_arguments(args)
         if method_name == 'to_string':
             if len(posargs) == 0:
-                if obj == True:
+                if obj is True:
```
Comment (jpakkane) [inline mesonbuild/interpreterbase.py]: Obj is always boolean here so the check (and the two other below) should be `if obj`.

Comment (ignatenkobrain) [inline mesonbuild/interpreterbase.py]: yeah, makes sense

File: mesonbuild/backend/ninjabackend.py (hunk 1 above)
Comment (QuLogic) [inline mesonbuild/backend/ninjabackend.py]: Put opening parenthesis on the line above and indent these by four spaces - no need for the backslash that way.

File: mesonbuild/backend/ninjabackend.py [hunk 3]
```diff
@@ -1274,10 +1274,10 @@ def generate_dynamic_link_rules(self, outfile):
 '''
                 else:
                     command_template = ' command = %s %s $ARGS  %s $in $LINK_ARGS $aliasing\n'
-                command = command_template % \
-                (' '.join(compiler.get_linker_exelist()),\
-                 ' '.join(cross_args),\
-                 ' '.join(compiler.get_linker_output_args('$out')))
+                command = (command_template %
+                    (' '.join(compiler.get_linker_exelist()),
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[flake8]__
* [E128] continuation line under-indented for visual indent

File: mesonbuild/backend/ninjabackend.py [hunk 4]
```diff
@@ -1449,12 +1449,12 @@ def generate_compile_rule_for(self, langname, compiler, qstr, is_cross, outfile)
 '''
         else:
             command_template = ' command = %s %s $ARGS %s %s %s $in\n'
-        command = command_template % \
-            (' '.join([ninja_quote(i) for i in compiler.get_exelist()]),\
+        command = (command_template %
+            (' '.join([ninja_quote(i) for i in compiler.get_exelist()]),
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[flake8]__
* [E128] continuation line under-indented for visual indent

File: mesonbuild/backend/ninjabackend.py [hunk 5]
```diff
@@ -1492,12 +1492,12 @@ def generate_pch_rule_for(self, langname, compiler, qstr, is_cross, outfile):
...
         else:
             output = ' '.join(compiler.get_output_args('$out'))
-        command = " command = %s %s $ARGS %s %s %s $in\n" % \
-            (' '.join(compiler.get_exelist()),\
-             ' '.join(cross_args),\
-             ' '.join(quoted_depargs),\
-             output,\
-             ' '.join(compiler.get_compile_only_args()))
+        command = (" command = %s %s $ARGS %s %s %s $in\n" %
+            (' '.join(compiler.get_exelist()),
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[flake8]__
* [E128] continuation line under-indented for visual indent

**Comment by msink** [comment]
Fixed E712, added rest of E502

**Comment by msink** [comment]
Next most common warnings are:
```
[E301] expected 1 blank line, found 0
[E303] too many blank lines (2)
[E401] multiple imports on one line
```

Need to be fixed or disabled?

**Comment by ignatenkobrain** [comment]
from my POV:

* E301 should be fixed
* E303 should be fixed
* E401 should be fixed (though it can take a while to decouple imports)

But this is just my vision, other members can disagree on this ;)

File: mesonbuild/environment.py [hunk 6]
```diff
@@ -12,14 +12,16 @@
...
-from .mesonlib import EnvironmentException, Popen_safe
 import configparser
+import os
+import platform
+import re
 import shutil
+import subprocess
+
+from . import coredata, mesonlib, mlog
+from .compilers import *
```
Comment (ignatenkobrain) [inline mesonbuild/environment.py]: __[flake8]__
* [F403] 'from compilers import *' used; unable to detect undefined names

File: mesonbuild/backend/ninjabackend.py [hunk 7]
```diff
@@ -1274,10 +1277,10 @@ def generate_dynamic_link_rules(self, outfile):
 '''
                 else:
                     command_template = ' command = %s %s $ARGS  %s $in $LINK_ARGS $aliasing\n'
-                command = command_template % \
-                (' '.join(compiler.get_linker_exelist()),\
-                 ' '.join(cross_args),\
-                 ' '.join(compiler.get_linker_output_args('$out')))
+                command = (command_template %
+                    (' '.join(compiler.get_linker_exelist()),
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[flake8]__
* [E128] continuation line under-indented for visual indent

File: mesonbuild/backend/ninjabackend.py [hunk 8]
```diff
@@ -1449,12 +1452,12 @@ def generate_compile_rule_for(self, langname, compiler, qstr, is_cross, outfile)
 '''
         else:
             command_template = ' command = %s %s $ARGS %s %s %s $in\n'
-        command = command_template % \
-            (' '.join([ninja_quote(i) for i in compiler.get_exelist()]),\
+        command = (command_template %
+            (' '.join([ninja_quote(i) for i in compiler.get_exelist()]),
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[flake8]__
* [E128] continuation line under-indented for visual indent

File: mesonbuild/backend/ninjabackend.py [hunk 9]
```diff
@@ -1492,12 +1495,12 @@ def generate_pch_rule_for(self, langname, compiler, qstr, is_cross, outfile):
...
         else:
             output = ' '.join(compiler.get_output_args('$out'))
-        command = " command = %s %s $ARGS %s %s %s $in\n" % \
-            (' '.join(compiler.get_exelist()),\
-             ' '.join(cross_args),\
-             ' '.join(quoted_depargs),\
-             output,\
-             ' '.join(compiler.get_compile_only_args()))
+        command = (" command = %s %s $ARGS %s %s %s $in\n" %
+            (' '.join(compiler.get_exelist()),
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[flake8]__
* [E128] continuation line under-indented for visual indent

File: mesonbuild/backend/ninjabackend.py (hunk 1 above)
Comment (QuLogic) [inline mesonbuild/backend/ninjabackend.py]: I meant move the opening parenthesis, not add another one.

**Comment by ignatenkobrain** [comment]
```python
  File "/root/mesonbuild/mesonmain.py", line 298, in run

    app.generate()

  File "/root/mesonbuild/mesonmain.py", line 182, in generate

    g.generate(intr)

  File "/root/mesonbuild/backend/ninjabackend.py", line 191, in generate

    self.generate_target(t, outfile)

  File "/root/mesonbuild/backend/ninjabackend.py", line 426, in generate_target

    obj_list.append(self.generate_single_compile(target, outfile, src, False, [], header_deps))

  File "/root/mesonbuild/backend/ninjabackend.py", line 1770, in generate_single_compile

    compiler)

  File "/root/mesonbuild/compilers.py", line 240, in get_base_compile_args

    args += compiler.get_colorout_args(options['b_colorout'].value)

  File "/root/mesonbuild/compilers.py", line 2106, in get_colorout_args

    if mesonlib.version_compare(self.version, '>=4.9.0'):

NameError: name 'mesonlib' is not defined
```

**Comment by jpakkane** [comment]
I personally don't like any of 301, 302 or 401. 402 is great.

**Comment by msink** [comment]
> I personally don't like any of 301, 302 or 401

I can remove them if you want, but as this project becomes more community driven
it should conform to most common style conventions. IMHO.

**Comment by jpakkane** [comment]
301 and 303 are harmless, but 401 is actively harmful, especially the alphabetizing thing (because it will, eventually, raise the pointless fight of whether they should be alphabetized or ordered by length or god nows what else).

**Comment by QuLogic** [comment]
I don't think pep8 or pycodestyle care about alphabetized imports; you'd have to use isort for that.

**Comment by msink** [comment]
Yes, E401 patch was done using autopep8 and isort.

**Comment by msink** [comment]
Almost done, remains things like this:
```
def detect_vcs(source_dir):
    vcs_systems = [
        dict(name = 'git',        cmd = 'git', repo_dir = '.git', get_rev = 'git describe --dirty=+', rev_regex = '(.*)', dep = '.git/logs/HEAD'),
        dict(name = 'mercurial',  cmd = 'hg',  repo_dir = '.hg',  get_rev = 'hg id -i',               rev_regex = '(.*)', dep = '.hg/dirstate'),
        dict(name = 'subversion', cmd = 'svn', repo_dir = '.svn', get_rev = 'svn info',               rev_regex = 'Revision: (.*)', dep = '.svn/wc.db'),
        dict(name = 'bazaar',     cmd = 'bzr', repo_dir = '.bzr', get_rev = 'bzr revno',              rev_regex = '(.*)', dep = '.bzr'),
    ]
```

pep8 complains here many times about:
```
[E251] unexpected spaces around keyword / parameter equals
[E241] multiple spaces after ','
```

Don't know how to fix, and should it be fixed.

**Comment by ignatenkobrain** [comment]
just disable those warnings for exact lines.

**Comment by msink** [comment]
Maybe better to disable E241 and E251 globally for now?

File: mesonbuild/coredata.py [hunk 10]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
-        'libdir':     [UserStringOption, 'Library directory.', default_libdir()],
-        'libexecdir': [UserStringOption, 'Library executable directory.', default_libexecdir()],
-        'bindir':     [UserStringOption, 'Executable directory.', 'bin'],
-        'sbindir':    [UserStringOption, 'System executable directory.', 'sbin'],
-        'includedir': [UserStringOption, 'Header file directory.', 'include'],
-        'datadir':    [UserStringOption, 'Data file directory.', 'share'],
-        'mandir':     [UserStringOption, 'Manual page directory.', 'share/man'],
-        'infodir':    [UserStringOption, 'Info page directory.', 'share/info'],
-        'localedir':  [UserStringOption, 'Locale data directory.', 'share/locale'],
+        'buildtype':  [UserComboOption, 'Build type to use.', ['plain', 'debug', 'debugoptimized', 'release', 'minsize'], 'debug'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 11]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'strip':      [UserBooleanOption, 'Strip targets on install.', False], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 12]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'unity':      [UserBooleanOption, 'Unity build.', False], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 13]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'prefix':     [UserStringOption, 'Installation prefix.', default_prefix()], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 14]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'libdir':     [UserStringOption, 'Library directory.', default_libdir()], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 15]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'libexecdir': [UserStringOption, 'Library executable directory.', default_libexecdir()], # noqa
+        'bindir':     [UserStringOption, 'Executable directory.', 'bin'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 16]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'sbindir':    [UserStringOption, 'System executable directory.', 'sbin'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 17]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'includedir': [UserStringOption, 'Header file directory.', 'include'], # noqa
+        'datadir':    [UserStringOption, 'Data file directory.', 'share'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 18]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'mandir':     [UserStringOption, 'Manual page directory.', 'share/man'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 19]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'infodir':    [UserStringOption, 'Info page directory.', 'share/info'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 20]
```diff
@@ -201,19 +205,19 @@ def get_builtin_option_default(optname):
...
+        'localedir':  [UserStringOption, 'Locale data directory.', 'share/locale'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 21]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
-        'localstatedir':   [UserStringOption, 'Localstate data directory.', 'var'],
-        'sharedstatedir':  [UserStringOption, 'Architecture-independent data directory.', 'com'],
-        'werror':          [UserBooleanOption, 'Treat warnings as errors.', False],
-        'warning_level':   [UserComboOption, 'Compiler warning level to use.', ['1', '2', '3'], '1'],
-        'layout':          [UserComboOption, 'Build directory layout.', ['mirror', 'flat'], 'mirror'],
-        'default_library': [UserComboOption, 'Default library type.', ['shared', 'static'], 'shared'],
-        'backend':         [UserComboOption, 'Backend to use.', backendlist, 'ninja'],
-        'stdsplit':        [UserBooleanOption, 'Split stdout and stderr in test logs.', True],
-        'errorlogs':       [UserBooleanOption, "Whether to print the logs from failing tests.", True],
+        'sysconfdir':      [UserStringOption, 'Sysconf data directory.', 'etc'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 22]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
+        'localstatedir':   [UserStringOption, 'Localstate data directory.', 'var'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 23]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
+        'sharedstatedir':  [UserStringOption, 'Architecture-independent data directory.', 'com'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 24]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
+        'werror':          [UserBooleanOption, 'Treat warnings as errors.', False], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 25]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
+        'warning_level':   [UserComboOption, 'Compiler warning level to use.', ['1', '2', '3'], '1'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 26]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
+        'layout':          [UserComboOption, 'Build directory layout.', ['mirror', 'flat'], 'mirror'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 27]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
+        'default_library': [UserComboOption, 'Default library type.', ['shared', 'static'], 'shared'], # noqa
+        'backend':         [UserComboOption, 'Backend to use.', backendlist, 'ninja'], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 28]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
+        'stdsplit':        [UserBooleanOption, 'Split stdout and stderr in test logs.', True], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/coredata.py [hunk 29]
```diff
@@ -223,16 +227,16 @@ def get_builtin_option_default(optname):
...
+        'errorlogs':       [UserBooleanOption, "Whether to print the logs from failing tests.", True], # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/coredata.py]: __[flake8]__
* [E241] multiple spaces after ':'

File: mesonbuild/mesonlib.py [hunk 30]
```diff
@@ -135,10 +140,10 @@ def exe_exists(arglist):
 
 def detect_vcs(source_dir):
     vcs_systems = [
-        dict(name = 'git',        cmd = 'git', repo_dir = '.git', get_rev = 'git describe --dirty=+', rev_regex = '(.*)', dep = '.git/logs/HEAD'),
-        dict(name = 'mercurial',  cmd = 'hg',  repo_dir = '.hg',  get_rev = 'hg id -i',               rev_regex = '(.*)', dep = '.hg/dirstate'),
-        dict(name = 'subversion', cmd = 'svn', repo_dir = '.svn', get_rev = 'svn info',               rev_regex = 'Revision: (.*)', dep = '.svn/wc.db'),
-        dict(name = 'bazaar',     cmd = 'bzr', repo_dir = '.bzr', get_rev = 'bzr revno',              rev_regex = '(.*)', dep = '.bzr'),
+        dict(name = 'git',        cmd = 'git', repo_dir = '.git', get_rev = 'git describe --dirty=+', rev_regex = '(.*)', dep = '.git/logs/HEAD'), # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/mesonlib.py]: __[flake8]__
* [E251] unexpected spaces around keyword / parameter equals
* [E241] multiple spaces after ','

File: mesonbuild/mesonlib.py [hunk 31]
```diff
@@ -135,10 +140,10 @@ def exe_exists(arglist):
...
+        dict(name = 'mercurial',  cmd = 'hg',  repo_dir = '.hg',  get_rev = 'hg id -i',               rev_regex = '(.*)', dep = '.hg/dirstate'), # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/mesonlib.py]: __[flake8]__
* [E251] unexpected spaces around keyword / parameter equals
* [E241] multiple spaces after ','

File: mesonbuild/mesonlib.py [hunk 32]
```diff
@@ -135,10 +140,10 @@ def exe_exists(arglist):
...
+        dict(name = 'subversion', cmd = 'svn', repo_dir = '.svn', get_rev = 'svn info',               rev_regex = 'Revision: (.*)', dep = '.svn/wc.db'), # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/mesonlib.py]: __[flake8]__
* [E251] unexpected spaces around keyword / parameter equals
* [E241] multiple spaces after ','

File: mesonbuild/mesonlib.py [hunk 33]
```diff
@@ -135,10 +140,10 @@ def exe_exists(arglist):
...
+        dict(name = 'bazaar',     cmd = 'bzr', repo_dir = '.bzr', get_rev = 'bzr revno',              rev_regex = '(.*)', dep = '.bzr'), # noqa
```
Comment (ignatenkobrain) [inline mesonbuild/mesonlib.py]: __[flake8]__
* [E251] unexpected spaces around keyword / parameter equals
* [E241] multiple spaces after ','

**Comment by msink** [comment]
Disabled flake8 warnings E241, E251, E305, E401
Now my local flake8 runs without any warnings.

Hope now it's ready for merging.

**Pull request merged**